backend/app/middlewares/loginAdminUser.js

```diff
@@ -0,0 +1,34 @@
+const authJwt = require("./authJwt");
+
+isAdmin = (req, res, next) => {
+  User.findOne({
+    username: req.body.username
+  }).exec((err, user) => {
+    if (err) {
+      res.status(500).send({ message: err });
+      return;
+    }
+
+    Role.find(
+      {
+        _id: { $in: user.roles }
+      },
+      (err, roles) => {
+        if (err) {
+          res.status(500).send({ message: err });
+          return;
+        }
+
+        for (let i = 0; i < roles.length; i++) {
+          if (roles[i].name === "admin") {
+            next();
+            return;
+          }
+          res.status(403).send({ message: "Forbidden" });
```

Comment:
Okay I get your question now. Let's move this one step out. Move it outside the loop